### ISSUE TITLE
Cloud Init not reading `drop-ins` sections since Flatcar 3975.2.0.

### ISSUE THREAD

**Issue by snikhil1998** [flatcar/Flatcar#1567]
## Description

CoreOS Cloud Init is not reading the `drop-ins` sections in `cloud-config.yml` since Flatcar 3975.2.0.

## Impact

Some fixes implemented by us are not being read from `cloud-config.yml`.

## Environment and steps to reproduce

1. **Set-up**: VMware ESXi
2. **Task**: Freshly deploying the OVA, modifications for `drop-ins` sections in `cloud-config.yml`

## Expected behavior

The `drop-ins` sections should be read and the drop-in files should be created/modified accordingly.

**Comment by jepio** [comment]
This is called out in the release notes for 3975.2.0:
"Removed coreos-cloudinit support for automatic keys conversion (e.g reboot-strategy -> reboot_strategy) ([scripts#1687](https://github.com/flatcar/scripts/pull/1687))"

We needed to remove this to upgrade the yaml library used in cloudinit, the previously used one was unmaintained and had vulnerabilities.

Changing it to `drop_ins` should work.

**Comment by snikhil1998** [comment]
Does changing `drop-ins` to `drop_ins` work on Flatcar Linux 3815.2.x and older?

**Comment by jepio** [comment]
Yes, the canonical name of that section was always `drop_ins` but the old yaml parser performed automatic conversion for `drop-ins`. This isn't supported in newer parser versions. It's also unfortunate that the examples in docs used `drop-ins`.
